src/mds/MDCache.cc

```diff
@@ -830,6 +840,23 @@ MDSCacheObject *MDCache::get_object(const MDSCacheObjectInfo &info)
 }
 
 
+// ====================================================================
+// consistent hash ring
+
+/*
+ * hashing implementation based on Lamping and Veach's Jump Consistent Hash: https://arxiv.org/pdf/1406.2294.pdf
```

Comment:
> based on

Did you make changes, or is it a verbatim copy? What changes did you make?

Reply:
Why is this algorithm useful for this work? Explain how it is used to implement consistent hashing for ephemeral pins.